src/components/ShareList.jsx

```diff
@@ -0,0 +1,99 @@
+import { shareList } from '../api/firebase';
+import { useState } from 'react';
+import { useAuth } from '../api';
+import { validateEmail } from '../utils/emailValidator.js';
+
+export function ShareList({ listPath }) {
+	const [recipientEmail, setRecipientEmail] = useState('');
+	const [emailValidator, setEmailValidator] = useState('');
+	const [isValid, setIsValid] = useState(false);
+	const { user } = useAuth();
+	const currentUserId = user?.uid;
+
+	const handleChange = (e) => {
+		setRecipientEmail(e.target.value);
+
+		if (validateEmail(e.target.value)) {
+			setEmailValidator('');
+			setIsValid(true);
+		} else {
+			setEmailValidator(
+				'Please enter a valid email, e.g., example@example.com.',
+			);
+		}
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		if (recipientEmail === '') {
+			return alert('Cannot submit without an email.');
+		}
+
+		if (!isValid) {
+			return alert('Cannot submit with an invalid email');
+		}
+
+		try {
+			await shareList(listPath, currentUserId, recipientEmail);
+			alert('List was successfully shared!');
+			setRecipientEmail('');
+		} catch (error) {
+			alert(error);
+		}
+	};
+
+	return (
+		<form onSubmit={handleSubmit}>
+			<p>You can share this shopping list with existing users!</p>
```

Comment:
This is out of scope, but I wanted to mention it in case you decide to implement it in the future. You can use semantic icons, which can be read by screen readers, to replace this with a little "share" icon in the future.